src/jit/hwintrinsicxarch.cpp

```diff
@@ -152,6 +159,61 @@ InstructionSet Compiler::isaOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].isa;
 }
 
+var_types Compiler::retTypeOfHWIntrinsic(NamedIntrinsic intrinsic)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].retType;
+}
+
+int Compiler::numParametersOfHWIntrinsic(NamedIntrinsic intrinsic)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numParameters;
+}
+
+int Compiler::ivalOfHWIntrinsic(NamedIntrinsic intrinsic)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].ival;
+}
+
+unsigned Compiler::simdSizeOfHWIntrinsic(NamedIntrinsic intrinsic)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].simdSize;
+}
+
+instruction Compiler::instructionOfHWIntrinsicOnType(NamedIntrinsic intrinsic, var_types type)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    assert(type >= TYP_BYTE && type <= TYP_DOUBLE);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].ins[type - TYP_BYTE];
+}
+
+HWIntrinsicCategory Compiler::categoryOfHWIntrinsic(NamedIntrinsic intrinsic)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].category;
+}
+
+GenTreeArgList* Compiler::buildArgList(int num)
```

Comment:
nit: This probably doesn't need to be a member on the compiler (it can probably just be a global static).

It also isn't really needed outside of the few intrinsics that take > 4 operands. The 3 and 4 operand encodings are common enough that having an explicit gtArgList constructor is useful (and probably more efficient)